File: backend-boot/src/main/java/com/oasis/oasis/model/TaskData.java
package com.oasis.oasis.model;

import java.util.Date;

public class TaskData {
    private Long id;
    private String title;
    private String taskId;
    private String description;
    private String category;
    private Date dueDate;
    private String priority;
    private Boolean completed;

    public Long getId() {
        return id;
    }
    public void setId(Long id) {
        this.id = id;
    }
    public String getTitle() {
        return title;
    }
    public void setTitle(String title) {
        this.title = title;
    }
    public String getTaskId() {
        return taskId;
    }
    public void setTaskId(String taskId) {
        this.taskId = taskId;
    }
    public String getDescription() {
        return description;
    }
    public void setDescription(String description) {
        this.description = description;
    }
    public String getCategory() {
        return category;
    }
    public void setCategory(String category) {
        this.category = category;
    }
    public Date getDueDate() {
        return dueDate;
    }
    public void setDueDate(Date dueDate) {
        this.dueDate = dueDate;
    }
    public String getPriority() {
        return priority;
    }
    public void setPriority(String priority) {
        this.priority = priority;
    }
    public Boolean getCompleted() {
        return completed;
    }
    public void setCompleted(Boolean completed) {
        this.completed = completed;
    }
}

File: frontend-angular/src/app/components/models/task_data.ts
import { User } from "./user";

export class TaskData {
    id: number;
    description: string;
    dueDate: Date;
    priority: string;
    category: string;
    title: string;
    taskId: string;
    completed: boolean;

    constructor() {
        this.id = 0;
        this.description = '';
        this.dueDate = new Date();
        this.priority = '';
        this.category = '';
        this.title = '';
        this.taskId = '';
        this.completed = false;
    }
}
